src/llama.cpp

```diff
@@ -4901,6 +4900,19 @@ static void llm_load_hparams(
                     case 40: model.type = e_model::MODEL_14B; break;
                     default: model.type = e_model::MODEL_UNKNOWN;
                 }
+
+                // for backward compatibility ; see: https://github.com/ggerganov/llama.cpp/pull/8931
+                if ((hparams.n_layer == 32 || hparams.n_layer == 40) && hparams.n_ctx_train == 4096) {
+                    // default value for Phi-3-mini-4k-instruct and Phi-3-medium-4k-instruct
+                    hparams.n_swa = 2047;
+                } else if (hparams.n_layer == 32 && hparams.n_head_kv(0) == 32 && hparams.n_ctx_train == 131072) {
+                    // default value for Phi-3-mini-128k-instruct
+                    hparams.n_swa = 262144;
+                } else if (hparams.n_layer == 40 && hparams.n_ctx_train == 131072) {
+                    // default value for Phi-3-medium-128k-instruct
+                    hparams.n_swa = 131072;
+                }
+                ml.get_key(LLM_KV_ATTENTION_SLIDING_WINDOW, hparams.n_swa, false);
```

Comment:
Might want to handle `get_key` returning `false` and `hparams.n_swa` remaining uninitialized to 0 by throwing an error, though probably never going to happen

Reply:
I initially thought that we can have the case you mentioned with a phi-3 small model, but turns out it's not supported yet. (ref: #7439)

So I agree with what you said. The check is added in https://github.com/ggerganov/llama.cpp/pull/8931/commits/2913e5ff1ef2c0c9a2cc7cc886f24f2dfaeda281